src/Components/CustomInput/index.tsx

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from "react"
+import { StyledInput } from "./StyledCom"
+import {MdSearch} from "react-icons/md"
+
+const CustomInput: React.FC<{type ?: string}> = ({type}) => {
+    const [types,setTypes] = useState<string>()
+    
+
+    useEffect(()=>{
+        type ? setTypes(type) : setTypes("text")
```

Comment:
ati ati ya make ternary gini, it is simple tp agak susah di baca
mending pake if statement aja untuk kasus ini